test_anagram (__main__.TestExercice) ... FAIL
test_doubles (__main__.TestExercice) ... ok
test_frequence (__main__.TestExercice) ... FAIL
test_grades (__main__.TestExercice) ... ERROR
test_order (__main__.TestExercice) ... ok

======================================================================
ERROR: test_grades (__main__.TestExercice)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_exercice.py", line 62, in test_grades
    output = exercice.best_grades(value)
  File "/home/runner/work/c04-ch6-exercices-rachiddeka/c04-ch6-exercices-rachiddeka/exercice.py", line 34, in best_grades
    best_student = dic
NameError: name 'dic' is not defined

======================================================================
FAIL: test_anagram (__main__.TestExercice)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_exercice.py", line 40, in test_anagram
    'Erreur dans la vérification de l\'anagramme'
AssertionError: Lists differ: [None, None] != [True, False]

First differing element 0:
None
True

- [None, None]
+ [True, False] : Erreur dans la vérification de l'anagramme

======================================================================
FAIL: test_frequence (__main__.TestExercice)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_exercice.py", line 83, in test_frequence
    'Erreur dans les histogrammes'
AssertionError: Lists differ: [{}, {}] != [{'B': 1, 'o': 4, 'n': 2, 'j': 2, 'u': 2, '[188 chars]: 1}]

First differing element 0:
{}
{'B': 1, 'o': 4, 'n': 2, 'j': 2, 'u': 2, 'r': 2, ',': 1, ' ': 1, 'b': 1}

- [{}, {}]
+ [{' ': 1, ',': 1, 'B': 1, 'b': 1, 'j': 2, 'n': 2, 'o': 4, 'r': 2, 'u': 2},
+  {' ': 5,
+   "'": 1,
+   '/': 1,
+   'J': 1,
+   'a': 6,
+   'b': 1,
+   'c': 3,
+   'd': 1,
+   'e': 7,
+   'f': 1,
+   'g': 2,
+   'i': 5,
+   'l': 3,
+   'm': 2,
+   'n': 4,
+   'o': 4,
+   'q': 1,
+   'r': 3,
+   't': 2,
+   'u': 2}] : Erreur dans les histogrammes

----------------------------------------------------------------------
Ran 5 tests in 0.001s

FAILED (failures=2, errors=1)
